src/lib/crypto/aes/operations.ts

```diff
@@ -0,0 +1,82 @@
+import { webcrypto } from 'one-webcrypto'
+import keys from './keys'
+import utils from '../utils'
+import { DEFAULT_SYMM_ALG, DEFAULT_CTR_LEN } from '../constants'
+import { SymmKey, SymmKeyOpts, SymmAlg, CipherText, Msg } from '../types'
+
+export async function encryptBytes(
+  msg: Msg,
+  key: SymmKey | string,
+  opts?: Partial<SymmKeyOpts>
+): Promise<CipherText> {
+  const data = utils.normalizeUtf16ToBuf(msg)
+  const importedKey = typeof key === 'string' ? await keys.importKey(key, opts) : key
+  const alg = opts?.alg || DEFAULT_SYMM_ALG
+  const iv = opts?.iv || utils.randomBuf(16)
+  const cipherBuf = await webcrypto.subtle.encrypt(
+    {
+      name: alg,
+      // AES-CTR uses a counter, AES-GCM/AES-CBC use an initialization vector
+      iv: alg === SymmAlg.AES_CTR ? undefined : iv,
+      counter: alg === SymmAlg.AES_CTR ? new Uint8Array(iv) : undefined,
+      length: alg === SymmAlg.AES_CTR ? DEFAULT_CTR_LEN : undefined,
+    },
+    importedKey,
+    data
+  )
+  return utils.joinBufs(iv, cipherBuf)
+}
+
+export async function decryptBytes(
+  msg: Msg,
+  key: SymmKey | string,
+  opts?: Partial<SymmKeyOpts>
+): Promise<ArrayBuffer> {
+  const cipherText = utils.normalizeBase64ToBuf(msg)
+  const importedKey = typeof key === 'string' ? await keys.importKey(key, opts) : key
+  const alg = opts?.alg || DEFAULT_SYMM_ALG
+  const iv = cipherText.slice(0, 16)
+  const cipherBytes = cipherText.slice(16)
```

Comment:
See the previous comment, but once there is a period separate this will need to be updated to split on it. We'll want to separately confirm the lengths of the two halves before we attempt to use them to catch garbage data early before doing any crypto work with them.

Reply:
done. check implemented as:
- is Iv 16 bytes?
- is the cipher text not 0 bytes, and cleanly divisible by 16 (AES-GCM block size in bytes)?